tests/pessimistic/pipeline/pipeline.go

```diff
@@ -0,0 +1,329 @@
+// Mostly, the return Response should implement UnknownResponse interface
+func (c *pipelineClient) Invoke(ctx context.Context, node cluster.ClientNode, r interface{}) core.UnknownResponse {
+	panic("not implemented")
+}
+
+// NextRequest generates a request for latter Invoke.
+func (c *pipelineClient) NextRequest() interface{} {
+	panic("not implemented")
+}
+
+// DumpState the database state(also the model's state)
+func (c *pipelineClient) DumpState(ctx context.Context) (interface{}, error) {
+	panic("not implemented")
+}
+
+// Start runs self scheduled cases
+// this function will block Invoke trigger
+// if you want to schedule cases by yourself, use this function only
+func (c *pipelineClient) Start(ctx context.Context, cfg interface{}, clientNodes []cluster.ClientNode) error {
+	c.scatterTable(dbName, tableName)
+	defer func() {
+		c.checkSum(context.Background())
+		c.reportSummary()
+		c.stopScatterTable(dbName, tableName)
+	}()
+
+	for i := 0; i < c.TableSize; i++ {
+		for j := 0; j < c.ContenderCount; j++ {
+			go c.transfer(ctx, i)
+		}
+	}
+
+	checkTick := time.NewTicker(time.Second)
+	scheduleTick := time.NewTicker(c.ScheduleInterval)
+	reportTick := time.NewTicker(c.ReportInterval)
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+
+		case <-checkTick.C:
+			c.checkSum(ctx)
+
+		case <-scheduleTick.C:
+			c.schedule(ctx)
+
+		case <-reportTick.C:
+			c.reportStats()
+		}
+	}
+}
+
+func (c *pipelineClient) scatterTable(db, table string) error {
+	url := fmt.Sprintf("http://%s/tables/%s/%s/scatter", c.dbStatusAddr, db, table)
+	_, err := httputil.NewHTTPClient(http.DefaultClient).Get(url)
+	return err
+}
+
+func (c *pipelineClient) stopScatterTable(db, table string) error {
+	url := fmt.Sprintf("http://%s/tables/%s/%s/stop-scatter", c.dbStatusAddr, db, table)
+	_, err := httputil.NewHTTPClient(http.DefaultClient).Get(url)
+	return err
+}
+
+func (c *pipelineClient) checkKeyIsLocked(ctx context.Context, row int) (bool, error) {
+	tx, err := c.db.Begin()
+	if err != nil {
+		return false, err
+	}
+	defer tx.Rollback()
+	rows, err := tx.QueryContext(ctx, fmt.Sprintf(`SELECT * FROM %s WHERE id in (%d, %d) FOR UPDATE NOWAIT`, tableName, row, row+1))
+	if err == nil {
+		rows.Close()
+		return false, nil
+	}
+	if me, ok := err.(*mysql.MySQLError); !ok || me.Number != 3572 {
+		return false, err
+	}
+	return true, nil
+}
+
+// transfer transfers 1 from a row to the next row.
+func (c *pipelineClient) transfer(ctx context.Context, row int) {
+	var cnt uint64
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		default:
+		}
+
+		tx, err := c.db.Begin()
+		if err != nil {
+			log.Fatalf("fail to begin a transaction: %v", err)
+		}
+		// When pipelined pessimistic locking is enabled, returning successful result means the lock request has been proposed.
+		if _, err := tx.ExecContext(ctx, fmt.Sprintf(`UPDATE %s SET v = v - 1 WHERE id = %d`, tableName, row)); err != nil {
+			log.Fatalf("fail to update: %v", err)
```

Comment:
I'm wondering if panic is suitable here. 

Usaually, panic means that the test case is failed and there maybe a bug related to **pipelined lock**.  However, we can't infer the conclusion from this error.

Reply:
If table `t` has multiple regions, I think we can enable `random-merge-scheduler` in nemesis.

Reply:
It shouldn't panic if we don't kill instances. It's possible a region is unavailable due to the change in raftstore. If don't panic here, we can't notice it.